lfsapi/schema/reader.go

```diff
@@ -0,0 +1,108 @@
+package schema
+
+import (
+	"bytes"
+	"io"
+	"strings"
+	"sync/atomic"
+
+	"github.com/git-lfs/git-lfs/errors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	// ErrNoValidationDone is an error returned when `ValidationErr()` is
+	// called before having processed any data.
+	ErrNoValidationDone = errors.New("lfsapi/schema: no validation done")
+	// ErrValidationIncomplete is an error returned when `ValidationErr()`
+	// is called while the reader is still processing data.
+	ErrValidationIncomplete = errors.New("lfsapi/schema: validation incomplete")
```

Comment:
Great call here. I made `ErrValidationIncomplete` be returned in place of `ErrNoValidationDone` in https://github.com/git-lfs/git-lfs/pull/1849/commits/721824deaacd1d64c7a7ff3f4103e80d1fce83ab. Also went ahead and un-exported the `ErrValidationIncomplete` variable in https://github.com/git-lfs/git-lfs/pull/1849/commits/640ca2134129ec4280b03c2bc0f5180ec885e09c.

I think this is ready to ship! 👍